File: source/com/barrybecker4/ui1/components/TexturedToolBar.java
// Copyright by Barry G. Becker, 2000-2011. Licensed under MIT License: http://www.opensource.org/licenses/MIT
package com.barrybecker4.ui1.components;

import com.barrybecker4.ui1.util.GUIUtil;

import javax.swing.*;
import java.awt.*;
import java.awt.event.ActionListener;

/**
 * a panel with a textured background.
 * The background gets tiled with the image that is passed in.
 */
public class TexturedToolBar extends JToolBar {
    private static final long serialVersionUID = 0L;
    private ImageIcon texture = null;

    protected static final Dimension MAX_BUTTON_SIZE = new Dimension( 100, 24 );

    /** the thing that processes the toolbar button presses. */
    protected ActionListener listener;


    public TexturedToolBar( ImageIcon texture, ActionListener listener ) {
        this.listener = listener;
        setTexture(texture);
    }

    public TexturedToolBar( ImageIcon texture) {
        setTexture(texture);
    }

    public void setListener(ActionListener listener ) {
        this.listener = listener;
    }

    public void setTexture( ImageIcon texture ) {
        this.texture = texture;
    }

    /**
     * create a toolbar button.
     */
    public GradientButton createToolBarButton( String text, String tooltip, Icon icon ) {
        GradientButton button = new GradientButton( text, icon );
        button.addActionListener(listener);
        button.setToolTipText( tooltip );
        button.setMaximumSize( MAX_BUTTON_SIZE );
        return button;
    }


    @Override
    public void paintComponent(Graphics g) {
        GUIUtil.paintComponentWithTexture(texture, this, g);
    }

}

File: scala-source/com/barrybecker4/ui/components/TexturedToolBar.scala
// Copyright by Barry G. Becker, 2017. Licensed under MIT License: http://www.opensource.org/licenses/MIT
package com.barrybecker4.ui.components

import com.barrybecker4.ui.util.GUIUtil
import javax.swing._
import java.awt._
import java.awt.event.ActionListener



object TexturedToolBar {
  protected val MAX_BUTTON_SIZE = new Dimension(100, 24)
}

/**
  * a panel with a textured background.
  * The background gets tiled with the image that is passed in.
  * @param theTexture the image for the texture background
  * @param theListener the thing that processes the toolbar button presses.
  * @author Barry Becker
  */
@SerialVersionUID(0L)
class TexturedToolBar(theTexture: ImageIcon, theListener: ActionListener) extends JToolBar {
  private var texture: ImageIcon = theTexture
  protected var listener: ActionListener = theListener


  def this(texture: ImageIcon) {
    this(texture, null)
  }

  def setListener(listener: ActionListener): Unit = {
    this.listener = listener
  }

  /** create a toolbar button. */
  def createToolBarButton(text: String, tooltip: String, icon: Icon): GradientButton = {
    val button = new GradientButton(text, icon)
    button.addActionListener(listener)
    button.setToolTipText(tooltip)
    button.setMaximumSize(TexturedToolBar.MAX_BUTTON_SIZE)
    button
  }

  override def paintComponent(g: Graphics): Unit = {
    GUIUtil.paintComponentWithTexture(texture, this, g)
  }
}